### PR TITLE
[IE/DML] Split conv2d to conv2d and convTranspose2d.

### PR DESCRIPTION
**Pull request by miaobin**
Split conv2d to conv2d and convTranspose2d. @fujunwei PTAL, thanks.

### CODE REVIEW

File: examples/SampleUtils.h [hunk 1]
```diff
@@ -99,11 +99,53 @@ namespace utils {
...
+            mOptions.activation = activation;
+
+            return &mOptions;
+        }
+
+      private:
+        ml::Conv2dOptions mOptions;
+    };
+
+    struct ConvTranspose2dOptions {
```
Comment (fujunwei) [inline examples/SampleUtils.h:133]: The struct `ConvTranspose2dOptions` and `Conv2dOptions` are similar, can we define a base template like
```
template<typename OptionType>
struct Conv2dBaseOptions {
   public:
        std::vector<int32_t> padding;
        std::vector<int32_t> strides;
        std::vector<int32_t> dilations;
        ml::AutoPad autoPad = ml::AutoPad::Explicit;
        int32_t groups = 1;
        ml::InputOperandLayout inputLayout = ml::InputOperandLayout::Nchw;
        ml::Operand bias;
        ml::Operator activation;

     const OptionType* AsPtr() {
     }

    private:
      OptionType mOptions;
}
```
Conv2d can extend it like
```
struct Conv2dOptions : pubilc ObjectBase<Conv2dBaseOptions> {
    pubilc:
       ml::Conv2dFilterOperandLayout filterLayout;
}
```
Transpose Conv2d can do like
```
struct ConvTranspose2dOptions: pubilc ObjectBase<Conv2dBaseOptions> {
    pubilc:
       ml::ConvTranspose2dFilterOperandLayout filterLayout;
        std::vector<int32_t> outputPadding;
        std::vector<int32_t> outputSizes;
}
```

Other implemenations like node/src/ops/ConvTranspose2d.cpp and src/webnn_native/ops/ConvTranspose2d.cpp also can consider this design.

Comment (miaobin) [reply examples/SampleUtils.h:133]: @fujunwei This is a very important reminder, I will update it in next commit.

**Comment by miaobin** [comment]
@fujunwei I have optimized the code. PTAL, thanks a lot!

File: src/webnn_native/ops/Conv2d.cpp [hunk 2]
```diff
@@ -91,33 +38,32 @@ namespace webnn_native { namespace op {
...
         auto inputShape = mInputs[0]->Shape();
         auto filterShape = mInputs[1]->Shape();
-
         bool nchw = mOptions.inputLayout == ml::InputOperandLayout::Nchw;
-        int32_t inputHeight = nchw ? inputShape[2] : inputShape[1];
-        int32_t inputWidth = nchw ? inputShape[3] : inputShape[2];
-        int32_t inputChannels = nchw ? inputShape[1] : inputShape[3];
 
+        int32_t batchSize = 0, inputChannels = 0, inputHeight = 0, inputWidth = 0;
+        utils::ParseInputOperand(mOptions.inputLayout, inputShape, batchSize, inputChannels,
```
Comment (fujunwei) [inline src/webnn_native/ops/Conv2d.cpp:44]: How about defining it like below:

std::vector<int32_t> GetInputShape(const OperandBase* input, ml::InputOperandLayout inputLayout) {
}
you can call it in L.39 with` auto inputShape = GetInputShape(mInputs[0], mOptions.inputLayout)`, another concern is why do you use the enum not` bool nchw` defined in L.41

**Review by fujunwei — COMMENTED** [review]
Some comments, PTAL, thanks.

File: src/webnn_native/ops/Conv2d.cpp (hunk 2 above)
Comment (miaobin) [reply src/webnn_native/ops/Conv2d.cpp]: Done and use the previous method to get input parameters.

**Comment by miaobin** [comment]
> Some comments, PTAL, thanks.

Updated the code according to our discussion. @fujunwei PTAL, thanks!

**Comment by fujunwei** [comment]
Merging, thanks.

**Pull request merged**